parsel/csstranslator.py

```diff
@@ -89,6 +101,116 @@ def xpath_text_simple_pseudo_element(self, xpath):
         """Support selecting text nodes using ::text pseudo-element"""
         return XPathExpr.from_xpath(xpath, textnode=True)
 
+    def xpath_first_of_pseudo(self, xpath):
```

Comment:
@eliasdorneles , no worries, I tried to be factual also. I hope I didn't sound offended or aggressive.
In fact I had to get back to [the specs](https://www.w3.org/TR/css3-selectors) to understand why I chose that way back then. So, good point.

As for the name, [`nth-of-type`](https://www.w3.org/TR/css3-selectors/#nth-of-type-pseudo) already exists, but are [not supported in cssselect](https://pythonhosted.org/cssselect/#supported-selectors)
We could try implementing it but it's for another PR perhaps.

Initially, I used `:nth()`, `:first` and `:last`. I still don't know what's best. We want to say:

> nth (resp. first, last) element matching previous predicates,

nth of what came before for that element, kinda